src/util/useId.ts

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+let id = 0;
+function genId() {
+  return ++id;
+}
+
+// Warning: probably not ssr friendly if using React <18
+export const useId =
+  // Prefer React's `useId` if it's available.
+  React.useId ??
+  function useId() {
+    const id = 'eds-' + genId();
```

Comment:
We could actually add a `console.warning` here as a reminder to anyone pulling this in. that could include any details we want to suggest, links to `useId` doc.s, etc.

Reply:
Sounds good, converted the two comments into one console warn with a link to the useId hook doc